### PR TITLE
SPCTR-3291: Fix - Image Gallery pagination loader issue

### PR DESCRIPTION
**Pull request by rajmohanreddy774**
### Description

When we select Masonry layout and enable image loader in pagination, even after all the images are loaded the image loader is still present.
https://brainstormforce.atlassian.net/browse/SPCTR-3291

### Screenshots

https://d.pr/v/bKgDCd

### Checklist:
- [x] My code is tested
- [ ] My code follows accessibility standards 
- [ ] My code has proper inline documentation 
- [ ] I've included any necessary tests 
- [ ] I've included developer documentation 
- [x] I've added proper labels to this pull request

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -215,6 +215,9 @@ Our external packages use [Rating Star Component](https://github.com/n49/react-s
 
 ## Changelog ##
 
+### x.x.x ###
+* Fix: Image Gallery - Resolved an issue in the masonry type wherein the loader is not hidden even after all the images are loaded.
```
Comment (bsf-zanev) [inline README.md:219]:
```suggestion
* Fix: Image Gallery - Resolved an issue with the masonry type wherein the loader would not be hidden after all the images have been loaded.
```

File: src/blocks/image-gallery/components/ImageGallery.js [hunk 2]
```diff
@@ -435,33 +435,37 @@ const ImageGallery = ( { attributes, setAttributes, name } ) => {
...
-						multiline={ false }
-						allowedFormats={ [ 'core/bold', 'core/italic', 'core/strikethrough' ] }
-					/>
-				</div>
-			) }
-		</>
-	);
+	const renderMasonryPagintion = () => {
+		if ( mediaGallery.length > paginateLimitFallback ) {
+			return (
```
Comment (bsf-zanev) [inline src/blocks/image-gallery/components/ImageGallery.js:440]: Can we avoid the big 'if' block of code?

Also remove the curly bracket of the 'if' when implementing this.

```suggestion
		if ( mediaGallery.length <= paginateLimitFallback ) {
			return null;
		}
		return(
```

**Review by bsf-zanev — COMMENTED** [review]
Code looks good, just suggested minor improvements and a changelog update.

File: src/blocks/image-gallery/components/ImageGallery.js (hunk 2 above)
Comment (bsf-zanev) [reply src/blocks/image-gallery/components/ImageGallery.js:440]: Although we could return this in ternary, that may be hard to read for developers that would pick up this block in the future.

Comment (rajmohanreddy774) [reply src/blocks/image-gallery/components/ImageGallery.js]: Implemented the suggestion.

File: README.md (hunk 1 above)
Comment (rajmohanreddy774) [reply README.md]: Done.

**Review by bsf-zanev — APPROVED** [review]
Looks good to me.

**Review by adi3890 — APPROVED** [review]

**Review by adi3890 — APPROVED** [review]

**Pull request merged**